File: week2/kuzey/EvaluateReversePolishNotation.java
class Solution {
    public int evalRPN(String[] tokens) {
        Stack<Integer> numbers = new Stack();

        for (String e : tokens) {
            switch (e) {
                case "/": {
                    int i1 = numbers.pop();
                    int i2 = numbers.pop();
                    numbers.push(i2 / i1);
                    break;
                }
                case "+": {
                    int i1 = numbers.pop();
                    int i2 = numbers.pop();
                    numbers.push(i2 + i1);
                    break;
                }
                case "-": {
                    int i1 = numbers.pop();
                    int i2 = numbers.pop();
                    numbers.push(i2 - i1);
                    break;
                }
                case "*": {
                    int i1 = numbers.pop();
                    int i2 = numbers.pop();
                    numbers.push(i2 * i1);
                    break;
                }
                default: {
                    numbers.push(Integer.parseInt(e));
                }
            }
        }

        return numbers.pop();
    }
}

File: week2/MarshalLee/ EvaluateReversePolishNotation.cs
using System;

public class EvaluateReversePolishNotation
{
    public int EvalRPN(string[] tokens)
    {
        Stack<int> stack = new();

        foreach (string token in tokens)
        {
            if (token == "+" || token == "-" || token == "*" || token == "/")
            {
                int secondOperand = stack.Pop();
                int firstOperand = stack.Pop();
                int result = token switch
                {
                    "+" => firstOperand + secondOperand,
                    "-" => firstOperand - secondOperand,
                    "*" => firstOperand * secondOperand,
                    "/" => firstOperand / secondOperand,
                    _ => throw new InvalidOperationException($"Unexpected token: {nameof(token)}")
                };
                stack.Push(result);
            }
            else
            {
                stack.Push(int.Parse(token));
            }
        }

        return stack.Pop();
    }
}
